### PR TITLE
Try to provide some more details on extension usage.

### PR DESCRIPTION
**Pull request by thomasvl**

### CODE REVIEW

File: Documentation/API.md [hunk 1]
```diff
@@ -641,36 +644,41 @@ extend CanBeExtended {
...
+* **Extension maps** are collections of extension objects indexed by the target
+  message and field number.  An extension map is generated for every file that
+  defined proto extensions and included as a static global variable.  It is
+  named based on the proto package, filename, and then ends in `_Extensions`,
+  so the above file would be `Sample_Extensions`. These maps are then used
+  the the `Message` apis for parsing/merging binary data to provide the
+  additional data to parse out the extension fields; if a mapping isn't
+  found, the extension field ends up in the `unknownFields` on the message.
+
+  If you need handle extension defined in multiple files, you can build up
```
Comment (allevato) [inline Documentation/API.md]: s/extension/extensions/

Comment (thomasvl) [reply Documentation/API.md]: Done.

File: Documentation/API.md [hunk 2]
```diff
@@ -641,36 +644,41 @@ extend CanBeExtended {
...
-  `ExtensibleMessage` and define some additional methods needed by the
-  other components.  You should not use these methods directly.
-* **Extension properties** use Swift's `extension` capability to add new properties to the
-  original message.  In many cases, you can simply use the extension properties
-  without understanding any of the other extension machinery.  The above
-  example creates a Swift extension of `CanBeExtended` that defines a new
-  property `extensionField` of type `Optional<Int32>`.
+  [`ExtensibleMessage`](https://github.com/apple/swift-protobuf/blob/master/Sources/SwiftProtobuf/ExtensibleMessage.swift)
+  and define some additional methods needed by the other components.  You
+  should _not_ need use these methods directly.
```
Comment (tbkka) [inline Documentation/API.md]: "... need to use ..."

File: Documentation/API.md [hunk 3]
```diff
@@ -641,36 +644,41 @@ extend CanBeExtended {
...
-    public subscript(messageType: Message.Type, protoFieldNumber: Int) -> MessageExtensionBase? {get}
-    public func fieldNumberForProto(messageType: Message.Type, protoFieldName: String) -> Int?
-}
-```
+* **Extension maps** are collections of extension objects indexed by the target
+  message and field number.  An extension map is generated for every file that
+  defined proto extensions and included as a static global variable.  It is
+  named based on the proto package, filename, and then ends in `_Extensions`,
+  so the above file would be `Sample_Extensions`. These maps are then used
+  the the `Message` apis for parsing/merging binary data to provide the
```
Comment (tbkka) [inline Documentation/API.md]: "... used by the ..." -- omitted word "by"

I suggest simplifying this sentence a bit: "... used by the `Message` apis for parsing/merging extension fields in the binary data; if a mapping ..."

File: Documentation/API.md [hunk 4]
```diff
@@ -641,36 +644,41 @@ extend CanBeExtended {
...
+  additional data to parse out the extension fields; if a mapping isn't
+  found, the extension field ends up in the `unknownFields` on the message.
+
+  If you need handle extensions defined in multiple files, you can build up
```
Comment (tbkka) [inline Documentation/API.md]: "If you need to handle..."  ---- omitted word "to"

File: Documentation/API.md (hunk 2 above)
Comment (thomasvl) [reply Documentation/API.md]: Done in #962

File: Documentation/API.md (hunk 3 above)
Comment (thomasvl) [reply Documentation/API.md]: done

File: Documentation/API.md (hunk 4 above)
Comment (thomasvl) [reply Documentation/API.md]: done

**Pull request merged**